napari/components/_viewer/model.py

```diff
@@ -353,12 +361,18 @@ def _calc_layers_ranges(self):
     def _calc_max_shape(self):
         """Calculates the max shape of all displayed layers.
         This assumes that all layers are stacked.
-        TODO: This is a temporary workaround until refactor is done
-        this method should not be used but instead '_calc_layers_ranges' should
-        be called.
         """
 
-        max_shape = [max-min for min, max, step in self._calc_layers_ranges()]
+        max_shape = [max for min, max, step in self._calc_layers_ranges()]
+
+        return max_shape
+
+    def _calc_min_shape(self):
```

Comment:
i think that min_shape and max_shape are poor names for what is going on here. from what i can tell, `min_shape` and `max_shape` really just describe the top-left and bottom-right corners of the viewer's bounding box

i also don't think that they should be in separate methods since they're related and calculated from the same method (`_calc_layers_ranges`)

i would just rename this to something like `_calc_bbox` and return a 2-tuple of 2-tuples equivalent to what is now `(min_shape, max_shape)`

Reply:
fixed